src/core/runtime/NOELLE_APIs.c

```diff
@@ -6,11 +6,20 @@ typedef struct {
 } DispatcherInfo;
 
 extern DispatcherInfo NOELLE_DOALLDispatcher(
-    void (*parallelizedLoop)(void *, int64_t, int64_t, int64_t),
+    void (*parallelizedLoop)(void *, int64_t, int64_t, int64_t, void *),
     void *env,
     int64_t maxNumberOfCores,
     int64_t chunkSize);
 
+extern int NOELLE_paraPrintfOneInt(int64_t coreID,
+                                   int64_t numCores,
+                                   int64_t chunkSize,
+                                   void *outputQueue,
+                                   char *format,
+                                   int arg);
+
+extern void NOELLE_Scylax_ChunkEnd(int8_t isChunkCompleted, void *outputQueue);
```

Comment:
I would consider making the name of this function more self-explanatory

Reply:
Agreed. Is chunking round-robin specific to DOALL?